### PR TITLE
vets website build inclusion for prod

### PR DESCRIPTION
**Pull request by mydesignrocks**
## Description
This is to include build within the auto deploy of prod and manual deploy of dev and staging
I will be testing the manual deploy of dev and staging first, before merging this

## Original issue(s)
department-of-veterans-affairs/va.gov-team#34836

## Testing done
tested dev: https://github.com/department-of-veterans-affairs/vets-website/actions/runs/1780507708

## Acceptance criteria
- [X] creates a build, validates the build within the deploy
- 
## Definition of done
- [X] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [X] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by jhonny218 — APPROVED** [review]

File: .github/workflows/daily-deploy-production.yml [hunk 1]
```diff
@@ -96,6 +97,39 @@ jobs:
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
 
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    needs: notify-start
```
Comment (theodur) [inline .github/workflows/daily-deploy-production.yml:103]:
```suggestion
    needs: validate-build-status
```
I think we only want the build job to run if the build status is successful. `notify-start` isn't dependent on `validate-build-status`

File: .github/workflows/daily-deploy-production.yml [hunk 2]
```diff
@@ -56,7 +57,7 @@ jobs:
       - name: Get commit sha (workflow_dispatch)
         if: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.commit_sha != '' }}
         run: echo 'COMMIT_SHA=${{ github.event.inputs.commit_sha }}' >> $GITHUB_ENV
-
+
```
Comment (theodur) [inline .github/workflows/daily-deploy-production.yml:60]:
```suggestion

```
It looks like some whitespaces were added here

File: .github/workflows/daily-deploy-production.yml [hunk 3]
```diff
@@ -96,6 +96,39 @@ jobs:
...
+    needs: validate-build-status
+    
+    steps:
+      - name: Build
+        run: yarn build --verbose —buildtype=vagovprod
+        timeout-minutes: 30
+
+      - name: Generate build details
+        run: |
+          cat > build/${{ BUILD_ENV }}/BUILD.txt << EOF
```
Comment (theodur) [inline .github/workflows/daily-deploy-production.yml:111]:
```suggestion
          cat > build/${{ env.BUILD_ENV }}/BUILD.txt << EOF
```

I think all uses of `BUILD_ENV` need to be prepended with `env` like this to access the environment variable

Comment (mydesignrocks) [reply .github/workflows/daily-deploy-production.yml:111]: yeah true that, let me adjust them all. thanks for pointing this too.

Comment (mydesignrocks) [reply .github/workflows/daily-deploy-production.yml]: made the changes.

**Comment by mydesignrocks** [comment]
I might have to do a couple more updates, still testing.

**Comment by mydesignrocks** [comment]
@theodur - this is now ready to be reviewed. thanks!

File: .github/workflows/daily-deploy-production.yml [hunk 4]
```diff
@@ -96,10 +97,58 @@ jobs:
...
+    name: Build
+    runs-on: self-hosted
+    needs: [set-env, notify-start, validate-build-status]
+    
+    env:
+      NODE_EXTRA_CA_CERTS: /etc/ssl/certs/VA-Internal-S2-RCA1-v1.cer.pem
+    
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (theodur) [inline .github/workflows/daily-deploy-production.yml:110]: We need to check out the correct sha for the build
```suggestion
        uses: actions/checkout@v2
        with:
          ref: ${{ needs.set-env.outputs.COMMIT_SHA }}
```

File: .github/workflows/daily-deploy-production.yml [hunk 5]
```diff
@@ -96,10 +97,60 @@ jobs:
...
+      - name: Generate build details
+        run: |
+          cat > build/${{ env.BUILD_ENV }}/BUILD.txt << EOF
+          BUILDTYPE=${{ env.BUILD_ENV }}
+          NODE_ENV=production
+          BRANCH_NAME=$(echo "${GITHUB_REF#refs/heads/}")
+          CHANGE_TARGET=null
+          RUN_ID=${{ github.run_id }}
+          RUN_NUMBER=${{ github.run_number }}
+          REF=${{ github.sha }}
```
Comment (theodur) [inline .github/workflows/daily-deploy-production.yml:136]:
```suggestion
          REF=${{ needs.set-env.outputs.COMMIT_SHA }}
```

**Review by theodur — APPROVED** [review]
I think we can also remove the [environment](https://github.com/department-of-veterans-affairs/vets-website/blob/master/.github/workflows/daily-deploy-production.yml#L155) variable in the `Deploy` job and use the new `BUILD_ENV` variable [here](https://github.com/department-of-veterans-affairs/vets-website/blob/master/.github/workflows/daily-deploy-production.yml#L189) instead of `${{ env.ENVIRONMENT }}`. Other than that, I think this looks good now!

**Comment by mydesignrocks** [comment]
> I think we can also remove the [environment](https://github.com/department-of-veterans-affairs/vets-website/blob/master/.github/workflows/daily-deploy-production.yml#L155) variable in the `Deploy` job and use the new `BUILD_ENV` variable [here](https://github.com/department-of-veterans-affairs/vets-website/blob/master/.github/workflows/daily-deploy-production.yml#L189) instead of `${{ env.ENVIRONMENT }}`. Other than that, I think this looks good now!

included that change, thanks!

**Pull request merged**